kolibri/auth/api.py

```diff
@@ -106,12 +106,14 @@ class FacilityUserViewSet(viewsets.ModelViewSet):
 
 class FacilityUsernameViewSet(viewsets.ReadOnlyModelViewSet):
     filter_backends = (filters.DjangoFilterBackend, filters.SearchFilter, )
-    queryset = FacilityUser.objects.filter(dataset__learner_can_login_with_no_password=True, roles=None).filter(
-        Q(devicepermissions__is_superuser=False) | Q(devicepermissions__isnull=True))
     serializer_class = FacilityUsernameSerializer
     filter_fields = ('facility', )
     search_fields = ('^username', )
 
+    def get_queryset(self):
+        return FacilityUser.objects.filter(dataset__learner_can_login_with_no_password=True, roles=None).filter(
+            Q(devicepermissions__is_superuser=False) | Q(devicepermissions__isnull=True))
+
```

Comment:
Moved this down into here because I was getting the following traceback:

```
(kolibri) jamalex@ace:~/fle/kolibri$ DJANGO_SETTINGS_MODULE=kolibri_instant_schools_plugin.instant_schools_settings kolibri --debug manage devserver 0.0.0.0:8080
Traceback (most recent call last):
  File "/home/jamalex/.virtualenvs/kolibri/bin/kolibri", line 9, in <module>
    load_entry_point('kolibri', 'console_scripts', 'kolibri')()
  File "/home/jamalex/fle/kolibri/kolibri/utils/cli.py", line 607, in main
    initialize(debug=debug)
  File "/home/jamalex/fle/kolibri/kolibri/utils/cli.py", line 192, in initialize
    django.setup()
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/jamalex/fle/kolibri/kolibri/core/models.py", line 5, in <module>
    registry.initialize()
  File "/home/jamalex/fle/kolibri/kolibri/plugins/registry.py", line 80, in initialize
    plugin_module = importlib.import_module(import_string)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/jamalex/fle/kolibri-instant-schools-plugin/kolibri_instant_schools_plugin/kolibri_plugin.py", line 4, in <module>
    from . import hooks, urls, about_urls, faq_urls
  File "/home/jamalex/fle/kolibri-instant-schools-plugin/kolibri_instant_schools_plugin/urls.py", line 4, in <module>
    from .api_urls import urlpatterns
  File "/home/jamalex/fle/kolibri-instant-schools-plugin/kolibri_instant_schools_plugin/api_urls.py", line 3, in <module>
    from .auth.api import PhoneNumberSignUpViewSet, PasswordResetTokenViewset, PhoneAccountProfileViewset
  File "/home/jamalex/fle/kolibri-instant-schools-plugin/kolibri_instant_schools_plugin/auth/api.py", line 7, in <module>
    from kolibri.auth.api import SignUpViewSet
  File "/home/jamalex/fle/kolibri/kolibri/auth/api.py", line 107, in <module>
    class FacilityUsernameViewSet(viewsets.ReadOnlyModelViewSet):
  File "/home/jamalex/fle/kolibri/kolibri/auth/api.py", line 109, in FacilityUsernameViewSet
    queryset = FacilityUser.objects.filter(dataset__learner_can_login_with_no_password=True, roles=None).filter(
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/manager.py", line 122, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/query.py", line 790, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/query.py", line 808, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1243, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1269, in _add_q
    allow_joins=allow_joins, split_subq=split_subq,
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1149, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1035, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1311, in names_to_path
    if field.is_relation and not field.related_model:
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/db/models/fields/related.py", line 113, in related_model
    apps.check_models_ready()
  File "/home/jamalex/.virtualenvs/kolibri/local/lib/python2.7/site-packages/django/apps/registry.py", line 131, in check_models_ready
    raise AppRegistryNotReady("Models aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Models aren't loaded yet.
```

After this change, everything worked swimmingly.

Reply:
Yeah, I guess otherwise the query is trying to get run at module import, before app registration?